electron/patch-desktop-filename.py

```diff
@@ -31,20 +34,26 @@
 import json
 import sys
 import os
+from pathlib import Path
 
 
 def main() -> None:
     asar_path = sys.argv[1]
-    extract_dir = tempfile.mktemp()
+    extract_dir = Path(tempfile.mktemp())
+    package_json_path = extract_dir / "package.json"
 
     asarPy.extract_asar(asar_path, extract_dir)
 
-    with open(os.path.join(extract_dir, "package.json"), "r", encoding="utf-8") as f:
+    with open(package_json_path, "r", encoding="utf-8") as f:
         package_json = json.load(f)
 
-    package_json["desktopName"] = os.getenv("CRAFT_PROJECT_NAME") + "_" + os.getenv("CRAFT_PROJECT_NAME") + ".desktop"
+    project_name = os.getenv("CRAFT_PROJECT_NAME")
+    if project_name is None:
+        exit(1)
```

Comment:
What if we raise an error here and then exit?

Reply:
Yeh, we can certainly print an error, give me a sec